memory/src/test/scala/filodb.memory/format/vectors/HistogramTest.scala

```diff
@@ -91,5 +91,40 @@ class HistogramTest extends NativeVectorTest {
       hist should not equal (hist2)
       hist.values shouldEqual addedBuckets
     }
+
+    it("should correctly add & makeMonotonic histograms containing NaNs") {
+      val hist1 = mutableHistograms(0).copy.asInstanceOf[MutableHistogram]
+      val histWNans = mutableHistograms(1)
+      histWNans.values(0) = Double.NaN
+      histWNans.values(2) = Double.NaN
+
+      hist1.add(histWNans)
+      hist1.makeMonotonic()
+      var current = 0d
+      hist1.valueArray.foreach { d =>
+        d should be >= (current)
+        current = d
+      }
+    }
+
+    it("should add histogram w/ diff bucket scheme and result in monotonically increasing histogram") {
+      val hist1 = mutableHistograms(0).copy.asInstanceOf[MutableHistogram]
+
+      val scheme2 = GeometricBuckets(2.0, 6.0, 3)
+      val hist2 = LongHistogram(scheme2, Array(10L, 20L, 40L))
+
+      hist1.add(hist2)
+      hist1.makeMonotonic()
```

Comment:
add() function should itself take care of calling makeMonotonic() if buckets are different.
I don't see any scenario where we wouldn't do that.

Reply:
Yes there is a scenario.  add() is used in sum_over_time within a chunk where we are guaranteed that buckets are the same, so no need for makeMonotonic().   However I can make the default add call that, and have an inner one which doesn't.